Take into account timeout limit when retrying queries (#46046)
rails/rails: activerecord/CHANGELOG.md

> *   Consider timeout limit when retrying a query

Reviewer: ```suggestion
*   Consider timeout limit when retrying a query.
```

    Building on the work done in #44576 and #44591, we extend the logic that automatically

>     reconnects broken db connections to take into account a timeout limit. We won't retry

Reviewer 2: I feel like broken is subjective so let's just say we reconnect.

```suggestion
    reconnects database connections to take into account a timeout limit. We won't retry
```

    a query if a given amount of time has elapsed since the query was first attempted. This

>     value will default to 5 seconds, but can be changes via the `connection_retry_timeout`

Reviewer 2: ```suggestion
    value will default to 5 seconds, but can be changed via the `connection_retry_timeout`
```
